### PR TITLE
fix(DATAGO-134691): improving artifacts appearing deleted during creation

### PR DESCRIPTION
**Pull request by lgh-solace**
### What is the purpose of this change?

Artifacts were appearing deleted when their content was not available during creation

### How was this change implemented?

This pull request refactors and improves the logic for detecting deleted artifacts in the chat UI, simplifies and standardizes styling for artifact bars, and adds targeted tests for the new utility logic. The main focus is on extracting artifact deletion logic into a dedicated utility function, updating components to use this utility, and improving maintainability and consistency in both code and styles.

**Artifact deletion detection improvements:**

* Extracted artifact deletion logic into a new utility function `isArtifactDeleted` in `artifactMessageUtils.ts`, clarifying the conditions under which an artifact is considered deleted and improving code reuse.
* Updated `ArtifactMessage.tsx` to use the new `isArtifactDeleted` utility for determining if an artifact is deleted, replacing the previous inline logic. [[1]](diffhunk://#diff-4f894c83d820b097c8790b10218a2c1f50dc2b9ebc1f22cd07bcc7239b7261ddR17) [[2]](diffhunk://#diff-4f894c83d820b097c8790b10218a2c1f50dc2b9ebc1f22cd07bcc7239b7261ddL77-R78)
* Added comprehensive unit tests for `isArtifactDeleted` in `artifactMessageUtils.test.ts`, covering all relevant edge cases.

**UI and style consistency:**

* Refactored `ArtifactBar.tsx` to use the `cn` utility for class name composition, standardized shadow and hover effects using new utility classes, and simplified the click handler logic.
* Improved and standardized sizing and flexbox utility classes in `ArtifactBar.tsx` for better maintainability and design consistency. [[1]](diffhunk://#diff-85650ae5acf047236bb0eeea861fb66ab2db8c7d92fafec17bde04e46c826872L204-R198) [[2]](diffhunk://#diff-85650ae5acf047236bb0eeea861fb66ab2db8c7d92fafec17bde04e46c826872L216-R208) [[3]](diffhunk://#diff-85650ae5acf047236bb0eeea861fb66ab2db8c7d92fafec17bde04e46c826872L244-R236)
* Added a `.card-surface-shadow` utility class in `index.css` for consistent shadow styling across components.

These changes collectively improve code clarity, maintainability, and UI consistency in the artifact-related components.

### How was this change tested?

- [x] Manual testing: [describe scenarios]
- [x] Unit tests: [new/modified tests]
- [ ] Integration tests: [if applicable]
- [ ] Known limitations: [what wasn't tested]

### Is there anything the reviewers should focus on/be aware of?

    Special attention areas, potential risks, or open questions

### CODE REVIEW

File: client/webui/frontend/src/lib/components/chat/artifact/ArtifactBar.tsx [hunk 1]
```diff
@@ -171,23 +171,15 @@ export const ArtifactBar: React.FC<ArtifactBarProps> = ({
...
-                }
-            }}
-            onClick={isDeleted ? undefined : handleBarClick}
+            className={cn(
+                "w-full bg-(--background-w10) transition-shadow duration-200 ease-in-out",
+                context === "list" ? "border-b" : "rounded",
+                isClickable && "cursor-pointer",
+                isDeleted && "opacity-60",
+                showShadow && "card-surface-shadow",
+                isClickable && "card-surface-hover"
```
Comment (lgh-solace) [inline client/webui/frontend/src/lib/components/chat/artifact/ArtifactBar.tsx:182]: Here, I'm trying to improve the style for the solaceDark theme which had the artifact bar nearly invisible - switching it to use the existing card class.

// DARK: before and after
Image


// LIGHT: before and after
Image

File: client/webui/frontend/src/lib/components/chat/artifact/ArtifactBar.tsx [hunk 2]
```diff
@@ -201,9 +193,9 @@ export const ArtifactBar: React.FC<ArtifactBarProps> = ({
             role={isClickable ? "button" : undefined}
             tabIndex={isClickable ? 0 : undefined}
         >
-            <div className="flex min-h-[60px] items-center gap-3 p-3">
+            <div className="flex min-h-15 items-center gap-3 p-3">
```
Comment (lgh-solace) [inline client/webui/frontend/src/lib/components/chat/artifact/ArtifactBar.tsx:187]: Just changes for lint warnings - nothing functional here.

File: client/webui/frontend/src/lib/components/chat/file/ArtifactMessage.tsx [hunk 3]
```diff
@@ -74,16 +75,7 @@ export const ArtifactMessage = (props: ArtifactMessageProps) => {
...
-    // However, don't mark as deleted if we have a valid fileAttachment with a URI -
-    // that means the artifact exists on the backend but just hasn't been fetched into the local list yet
-    const isDeleted = useMemo(() => {
-        if (props.status !== "completed") return false;
-        if (artifactInAll) return false; // Found in list (including hidden), not deleted
-        // If we have a fileAttachment with a URI, the artifact exists on backend (just not fetched yet)
-        if (fileAttachment?.uri) return false;
-        return true; // Completed, not in list, no URI = likely deleted
-    }, [props.status, artifactInAll, fileAttachment?.uri]);
+    const isDeleted = useMemo(() => isArtifactDeleted({ status: props.status, artifactInfo: artifactInAll, fileAttachment, message: props.message }), [props.status, artifactInAll, fileAttachment, props.message]);
```
Comment (lgh-solace) [inline client/webui/frontend/src/lib/components/chat/file/ArtifactMessage.tsx:78]: Function extracted for testing.

File: client/webui/frontend/src/lib/index.css [hunk 4]
```diff
@@ -40,8 +40,6 @@
 
 :root {
     --radius: 0.5rem;
-    /* Semantic theme variables are set by ThemeProvider via inline styles */
-    --tw-shadow-colored: 0 4px 6px -1px var(--primary-w90), 0 2px 4px -2px var(--primary-w90);
```
Comment (lgh-solace) [inline client/webui/frontend/src/lib/index.css:44]: Unused.

**Review by amir-ghasemi — APPROVED** [review]
LGTM. Thanks for the fix!
